### ISSUE TITLE
[Bug] Grafana doesn't load data when setting max-select-buckets to non-zero value

### ISSUE THREAD

**Issue by wei-hai** [influxdata/influxdb#8623]
__System info:__
InfluxDB: 1.3.1
OS: Mac OS

__Steps to reproduce:__
influxdb.conf
```
[cluster]
  shard-writer-timeout = "5s"
  write-timeout = "10s"
  max-concurrent-queries = 0
  query-timeout = "0s"
  max-select-point = 0
  max-select-series = 0
  max-select-buckets = 1000000
```

__Expected behavior:__
Grafana shall display data
__Actual behavior:__
Grafana shows `no data points`

`max-select-buckets = 0`
<img width="1279" alt="screen shot 2017-07-24 at 12 51 55 pm" src="https://user-images.githubusercontent.com/2936715/28541773-053336aa-706f-11e7-897c-8878d5041e5e.png">

`max-select-buckets = 1000000`
<img width="1271" alt="screen shot 2017-07-24 at 12 51 35 pm" src="https://user-images.githubusercontent.com/2936715/28541781-0b8aecbe-706f-11e7-9944-af1591871b84.png">

**Comment by sqreek** [comment]
we felt down in same problem.
It seems that range provided from kibana is not closed.
If you add and time < now() it will works
https://github.com/influxdata/influxdb/blob/253b8c2b35f8529fe597d065143c0c2e287a5b15/query/compile.go#L823
max range in such situation could be solved on influx side (by using now as default for max) or on kibana side (add time< now())

**Comment by rbetts** [comment]
We have users that write future (forecasted) data into the database and then overwrite it with current data as current data arrives. I think implying `now()` as a max time would break that use case. I'd prefer the query engine not guess the user's intentions and imply a `max`.

**Comment by jsternberg** [comment]
This should likely treat `now()` as the implicit end. We do that somewhere in the query engine when there is a `GROUP BY time(...)` in the query. If there's no interval, then the implicit end time is the maximum possible timestamp.

The bucket check must happen before that time gets reset. I'll take a look.

**Comment by jsternberg** [comment]
I took a look. This appears to be fixed in master, but the bug exists in 1.3. The fix for 1.3 is different than the one in master because it got fixed in master by accident during a refactor of the code.
